src/components/InventoryManager/ListingInputField.js

```diff
@@ -1,92 +1,264 @@
 
 const useStyles = makeStyles({
-  inputField: {
-    width: '100%',
-    height: '100%',
+  root: {
+    fontFamily: 'Work Sans',
+  },
+  heading: {
+    fontFamily: 'Work Sans',
+    fontWeight: 'bold',
+    fontSize: '1.4rem',
+    textDecoration: '#53AA48 solid underline 2px',
+    textUnderlinePosition: 'under',
+    textUnderlineOffset: '0.1rem',
+  },
+  subheading: {
+    fontFamily: 'Work Sans',
+    fontWeight: '600',
+    fontSize: '1rem',
+  },
+  text: {
+    fontFamily: 'Work Sans',
+    fontSize: '1.2rem',
+  },
+  numberInputField: {
+    maxWidth: '3.6em',
+    fontFamily: 'Work Sans',
+    padding: '0',
+    fontSize: '1.2rem',
+  },
+  infoLabel: {
+    fontWeight: 700,
+    fontFamily: 'Work Sans',
+    fontSize: '1.1rem',
+    textDecoration: '#53AA48 solid underline 2px',
+    textUnderlinePosition: 'under',
+    textUnderlineOffset: '0.1rem',
+  },
+  button: {
+    color: '#53AA48',
   },
 });
 
 export default function ListingInputField({
-  id, name, label, type, val, onChange, multiline,
+  id, name, label, type, val, onChange, options, getLabel,
+  placeholder, onButtonClick,
 }) {
   const classes = useStyles();
+  const textFieldRef = useRef();
   function getInputProps() {
-    let InputProps = {};
-    if (type === 'number' && label === 'units per pallet') {
+    let InputProps = {
+      classes: {
+        input: classes.text,
+      },
+    };
+    if (type === 'number') {
       InputProps = {
+        ...InputProps,
         inputProps: {
+          style: {
+            textAlign: 'center',
+          },
           min: 1,
-          max: 24,
-        },
-      };
-    } else if (type === 'number') {
-      InputProps = {
-        inputProps: {
-          min: 0,
         },
       };
     }
     return InputProps;
   }
   const inputProps = getInputProps({});
-  if (multiline) {
+  if (type === 'currency') {
+    return (
+      <>
+        <CurrencyTextField
+          variant="outlined"
+          placeholder="0.00"
+          currencySymbol="$"
+          value={val}
+          decimalCharacter="."
+          onChange={onChange}
+          textAlign="left"
+          size="small"
+        />
+      </>
+    );
+  }
+  if (type === 'autoComplete') {
     return (
       <>
-        <Typography variant="h6" component="h6">{label}</Typography>
+        <Grid item xs={12}>
+          <Typography className={classes.infoLabel} variant="h2" component="h2">{label}</Typography>
+        </Grid>
+        <Grid item xs={12}>
+          <Autocomplete
+            autoComplete
+            options={options}
+            getOptionLabel={getLabel}
+            onChange={onChange}
+            value={val}
+            renderInput={(params) => (
+              <TextField
+                name={name}
+                {...params}
+                placeholder={placeholder}
+                variant="outlined"
+                required
+              />
+            )}
+            classes={{ input: classes.text }}
```

Comment:
classes applies to the inner component, so it applies to the TextField component rendered by the AutoComplete.
I couldnt use className in the textfield though since it would overwrite the styling the AutoComplete has for it